### PR TITLE
Wheel issue and pkg_resource refactor

### LINKED ISSUE #146: installation: Wheel not seen by tensorboard
**Describe the bug**
Wheel gets successfully installed via conda but is not seen by all pip packages

**To Reproduce**
Follow GPU or CPU installation

**Expected behavior**

**Screenshots**
```sh
  File "/u/jab/code/gt4sd-core/src/gt4sd/cli/inference.py", line 35, in <module>
    from ..algorithms.registry import ApplicationsRegistry
  File "/u/jab/code/gt4sd-core/src/gt4sd/algorithms/__init__.py", line 28, in <module>
    from .conditional_generation.guacamol.core import (  # noqa: F401
  File "/u/jab/code/gt4sd-core/src/gt4sd/algorithms/conditional_generation/guacamol/__init__.py", line 26, in <module>
    from .core import (
  File "/u/jab/code/gt4sd-core/src/gt4sd/algorithms/conditional_generation/guacamol/core.py", line 28, in <module>
    from ....training_pipelines.core import TrainingPipelineArguments
  File "/u/jab/code/gt4sd-core/src/gt4sd/training_pipelines/__init__.py", line 30, in <module>
    import pkg_resources
  File "/u/jab/.conda/envs/gt4sd/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3242, in <module>
    @_call_aside
  File "/u/jab/.conda/envs/gt4sd/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3226, in _call_aside
    f(*args, **kwargs)
  File "/u/jab/.conda/envs/gt4sd/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3255, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/u/jab/.conda/envs/gt4sd/lib/python3.7/site-packages/pkg_resources/__init__.py", line 570, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/u/jab/.conda/envs/gt4sd/lib/python3.7/site-packages/pkg_resources/__init__.py", line 583, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/u/jab/.conda/envs/gt4sd/lib/python3.7/site-packages/pkg_resources/__init__.py", line 772, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'wheel>=0.26' distribution was not found and is required by tensorboard
```

### PR DESCRIPTION
**Pull request by jannisborn**
Closes #146 

This PR:
- makes `scipy` an explicit dependencies of the PyPI package since it's used directly inside GFlowNets and it removes `tensorflow` from that list since it's not used directly anywhere. This removal by itself already solved #146
- I took the opportunity to migrate from `pkg_resources` to importlib, since in #146 `pkg_resources` was complaining. This was done also because use of `pkg_resources` is discouraged in favor of `importlib`. Note that we install `importlib_metadata` and `importlib_resources` from PyPI which are backports of the standardlibrary packages (`importlib.*`) which are only available for Python 3.8+. I did this to support python 3.7 and 3.8+ without conditional imports

### CODE REVIEW

File: src/gt4sd/training_pipelines/tests/test_training_moses_vae.py [hunk 1]
```diff
@@ -23,19 +23,21 @@
...
-import pkg_resources
+import importlib_resources
 
 from gt4sd.training_pipelines import TRAINING_PIPELINE_MAPPING, MosesVAETrainingPipeline
 
-TEST_DATA_DIRECTORY = pkg_resources.resource_filename(
-    "gt4sd",
-    "training_pipelines/tests/",
-)
+file_manager = ExitStack()
```
Comment (drugilsberg) [inline src/gt4sd/training_pipelines/tests/test_training_moses_vae.py:37]: Let's extract a method as the only thing that is changing is the file path

File: src/gt4sd/training_pipelines/__init__.py [hunk 2]
```diff
@@ -208,16 +210,15 @@ def training_pipeline_name_to_metadata(name: str) -> Dict[str, Any]:
...
     if name in TRAINING_PIPELINE_NAME_METADATA_MAPPING:
         try:
-            with open(
-                pkg_resources.resource_filename(
-                    "gt4sd",
-                    f"training_pipelines/{TRAINING_PIPELINE_NAME_METADATA_MAPPING[name]}",
-                ),
-                "rt",
-            ) as fp:
+            ref = importlib_resources.files("my.package") / "resource.dat"
```
Comment (drugilsberg) [inline src/gt4sd/training_pipelines/__init__.py:219]: I really struggle to follow the logic here, maybe it's a problem of copy-pasting leftovers.

Comment (drugilsberg) [reply src/gt4sd/training_pipelines/__init__.py:219]: I would consider also using here the method we create for usage in the tests.

**Review by drugilsberg — CHANGES_REQUESTED** [review]
Amazing work really, there are only two comments to address.

File: src/gt4sd/training_pipelines/__init__.py (hunk 2 above)
Comment (jannisborn) [reply src/gt4sd/training_pipelines/__init__.py]: ah great catch, indeed a copy-paste leftover. Strange the tests passed anyways

File: src/gt4sd/training_pipelines/tests/test_training_moses_vae.py (hunk 1 above)
Comment (jannisborn) [reply src/gt4sd/training_pipelines/tests/test_training_moses_vae.py]: Done

File: src/gt4sd/conftest.py [hunk 3]
```diff
@@ -22,11 +22,33 @@
...
 import pytest
 
 
 @pytest.fixture
 def mock_wrong_s3_env(monkeypatch):
     """Changes an environment variable to break the s3 connection."""
     monkeypatch.setenv("GT4SD_S3_SECRET_KEY", "(╯°□°）╯︵ ┻━┻")
+
+
+def exitclose_file_creator(file_path: str) -> PosixPath:
```
Comment (drugilsberg) [inline src/gt4sd/conftest.py:39]: I would move this to `src/gt4sd/tests/utils.py` as this file was supposed to contain only pytest fixtures.

**Review by drugilsberg — COMMENTED** [review]
LGTM, only a minor comment left

**Pull request merged**